interpolate/d2v-interpolate-common/src/main/kotlin/io/data2viz/interpolate/basis.kt

```diff
@@ -17,12 +18,12 @@ fun computeSpline(t1: Double, v0: Int, v1: Int, v2: Int, v3: Int): Double {
 /**
  * uniform nonrational B-spline interpolation
  */
-fun basis(values: List<Int>): (Double) -> Double {
+fun basis(values: List<Int>): Interpolator<Double> {
     val n = values.size - 1
-    return fun(t: Double): Double {
+    return fun(t: Percent): Double {
 
-        val newT = t.coerceIn(0.0, 1.0)
-        val currentIndex: Int = if (t <= 0) 0 else if (t >= 1) n - 1 else floor(t * n).toInt()
+        val newT = t.coerceToDefault().value
```

Comment:
Create `val` or `var` the closest to its use. It helps when reading code. In this case, `newT` is used only one time. It could be inlined.